snapcraft/internal/elf.py

```diff
@@ -89,8 +89,25 @@ def __init__(self, *, soname: str, path: str, root_path: str,
             self.in_base_snap = False
 
 
+_soname_paths = dict()  # type: Dict[str, str]
```

Comment:
Global state tends to bite us. Would this be better as a class instead?

Reply:
I can work something out